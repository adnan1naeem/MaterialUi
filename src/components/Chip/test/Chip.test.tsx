import { Chip } from '../';
import renderer from 'react-test-renderer';
import React from 'react';
import 'jest';
import console = require('console');

test('Chip component with default values', () => {
  const chip = renderer.create(
    <Chip label={'10 % off'}>
    </Chip>
  ).toJSON();
  console.log('Chip', chip);
  expect(chip ? chip.props.style : null).toEqual(undefined);
  expect(chip).toMatchSnapshot();
});

test('Chip component with style', () => {
  const chip = renderer.create(
    <Chip style={{ 'background-color': 'green' }} label={'10 % off'}>
    </Chip>
  ).toJSON();
  expect(chip ? chip.props.style : null).toEqual({ 'background-color': 'green' });
  expect(chip).toMatchSnapshot();
});
