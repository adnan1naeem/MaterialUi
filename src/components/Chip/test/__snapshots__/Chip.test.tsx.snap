// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip component with default values 1`] = `
<div
  className="MuiChip-root-1"
  onClick={undefined}
  onKeyDown={[Function]}
  role="button"
  tabIndex={-1}
>
  <span
    className="MuiChip-label-6"
  >
    10 % off
  </span>
</div>
`;

exports[`Chip component with style 1`] = `
<div
  className="MuiChip-root-1"
  onClick={undefined}
  onKeyDown={[Function]}
  role="button"
  style={
    Object {
      "background-color": "green",
    }
  }
  tabIndex={-1}
>
  <span
    className="MuiChip-label-6"
  >
    10 % off
  </span>
</div>
`;
